bigc/api_client.py

```diff
@@ -215,3 +211,47 @@ def get_many(
                 raise TypeError(f"expected list, got {type(res_data['data']).__name__}")
 
             yield from res_data['data']
+
+    def _get_many_using_cursor(
+        self,
+        path: str,
+        *,
+        page_size: int,
+        params: dict[str, Any],
+        timeout: float | None,
+    ) -> Iterator[Any]:
+        if params.keys() & {'limit', 'before', 'after'}:
+            raise ValueError('params already has pagination values (limit, before, and/or after)')
+
+        params['limit'] = page_size
+
+        while True:
+            response = super().request('GET', path, params=params, timeout=timeout)
+
+            yield from response['data']
+
+            if not (
+                # end_cursor will still be provided if the next page is empty
+                response['meta']['cursor_pagination']['links'].get('next')
+                or response['data']
+            ):
```

Comment:
Shouldn't this be:

```py
if (
    not response['meta']['cursor_pagination']['links'].get('next')
    or len(response['data']) == 0
)
```

Right now, it's saying we stop if there is no next page _and_ no data on the current page, when I would think we would want to stop if either of those conditions is reached.

Reply:
`(not a) or (not b)` should be `not (a and b)`, you have `not (a or b)`.

Reply:
https://github.com/MedShift/bigc/pull/57/commits/2302c1440b14035b42d548360d7727c7d6b2dffe